plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -0,0 +1,197 @@
+                                RowType.field("job_name", VARCHAR),
+                                RowType.field("run_id", VARCHAR),
+                                RowType.field("job_owner_id", VARCHAR),
+                                RowType.field("trigger_type", VARCHAR))))
+                        .add(new ColumnMetadata("notebook", RowType.rowType(
+                                RowType.field("notebook_id", VARCHAR))))
+                        .add(new ColumnMetadata("cluster_id", VARCHAR))
+                        .add(new ColumnMetadata("read_version", BIGINT))
+                        .add(new ColumnMetadata("isolation_level", VARCHAR))
+                        .add(new ColumnMetadata("is_blind_append", BOOLEAN))
+                        //TODO add support for operationMetrics, userMetadata, engineInfo
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (transactionLogEntries.isEmpty()) {
+            return new FixedPageSource(ImmutableList.of());
+        }
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+        TimeZoneKey timeZoneKey = session.getTimeZoneKey();
+
+        transactionLogEntries.forEach(transactionLogEntry -> {
+            pagesBuilder.beginRow();
+
+            pagesBuilder.appendBigint(transactionLogEntry.getCommitInfo().getVersion());
+            pagesBuilder.appendTimestampTzMillis(transactionLogEntry.getCommitInfo().getTimestamp(), timeZoneKey);
+            write(transactionLogEntry.getCommitInfo().getUserId(), pagesBuilder);
+            write(transactionLogEntry.getCommitInfo().getUserName(), pagesBuilder);
+            write(transactionLogEntry.getCommitInfo().getOperation(), pagesBuilder);
+            if (checkNonNull(transactionLogEntry.getCommitInfo().getOperationParameters(), pagesBuilder)) {
+                pagesBuilder.appendVarcharVarcharMap(transactionLogEntry.getCommitInfo().getOperationParameters());
+            }
+            if (checkNonNull(transactionLogEntry.getCommitInfo().getJob(), pagesBuilder)) {
+                writeCommitInfoEntryJobBlock(pagesBuilder.nextColumn(), transactionLogEntry.getCommitInfo().getJob());
+            }
+            if (checkNonNull(transactionLogEntry.getCommitInfo().getNotebook(), pagesBuilder)) {
+                writeCommitInfoEntryNotebookBlock(pagesBuilder.nextColumn(), transactionLogEntry.getCommitInfo().getNotebook());
```

Comment:
I would rather keep it as it is.
Yes, it could be done a bit more generic - but in the current context i find it is better for readability to keep things as they are.